src/txn/txn.c

```diff
@@ -1768,9 +1846,24 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
         case WT_TXN_OP_BASIC_ROW:
         case WT_TXN_OP_INMEM_COL:
         case WT_TXN_OP_INMEM_ROW:
-            upd = op->u.op_upd;
-
             if (!prepare) {
+                /*
+                 * The page carrying the updates could have been evicted already. We will need to
+                 * fix the update pointer in the transaction op in such a case. If the resolution
+                 * returns a reference, the page was not evicted and we hold a strong hazard pointer
+                 * that we need to clear after the use. For testing purposes assert if resolution
+                 * fails.
+                 */
+                ret = __txn_resolve_weak_hazard_updates(session, op, &cursor, &ref);
+                WT_ERR_ASSERT(
+                  session, ret == 0, ret, "Failed to resolve an update during txn_commit");
+
+                if (ref == NULL)
+                    slow_resolved = true;
+                else
+                    WT_WITH_BTREE(session, op->btree, ret = __wt_hazard_clear(session, ref));
```

Comment:
Isn't it a requirement to clear the hazard pointer after properly "committing" the transaction, until the transaction is fully committed and its id has been cleared I think eviction could chose these updates and evict them flagging them as "uncommitted"?

Reply:
You are right, let's handle this in future changes. I have added a comment explaining that we will need to take care of the possibility of updates getting evicted before txn commits/rollbacks.